src/libraries/Directory.Build.targets

```diff
@@ -286,4 +286,28 @@
     </When>
   </Choose>
 
+  <!--
+    SDK currently does not know about net6.0 TargetFramework.
+    We need to manually add these to correctly resolve the targeting packs.
+    https://github.com/dotnet/sdk/issues/13950  
+  -->
+  <ItemGroup>
+    <KnownFrameworkReference Include="Microsoft.NETCore.App"
```

Comment:
I think we should instead figure out why we need all this. This is causing issues depending on the SDK's installed on the machine. https://github.com/dotnet/runtime/issues/45755

Reply:
These items need to be set for any project that doesn't set `DisableImplicitFrameworkReferences` to true and uses a `TargetFramework` that the SDK doesn't understand yet.

Reply:
I agree with Santi that we should not define them globally but only where we need them in a sharable manner.

Reply:
I think we need it in all of Libraries in order to target 6.0.  In libraries it should never cause the error (see my thread above where we set the property to disable that error).

Outside of libraries I agree we need to make this opt in granular.  All those places are layering problems where we're building against an LKG instead of live bits and represent problems with product layering.   We'll lose sight of those places once we have a newer SDK.

Reply:
> I think we need it in all of Libraries in order to target 6.0. In libraries it should never cause the error (see my thread above where we set the property to disable that error).

Why do we need this in all of libraries if we already use the live built targeting pack there?

> We'll lose sight of those places once we have a newer SDK.

I agree. That said, there aren't plans yet to update to a 6.0 SDK anytime soon.

Reply:
> Why do we need this in all of libraries if we already use the live built targeting pack there?

Because of the way you update the resolved targeting pack to point to a local directory the SDK needs to be able to resolve a `KnownFrameworkReference` for the tfm you're building.

Reply:
Sorry, I should marked the conversation as resolved. This isn't needed for all of libraries, only for parts which use framework references (anything not inbox). I fixed this with https://github.com/dotnet/runtime/commit/adad810e664f53211aed98f0f06b6beeaede180c.